### PR TITLE
1801 Mod 1 Final Independent Assessment

### PR DESCRIPTION
**Pull request by Shniks**
Completed Iteration 3.

### CODE REVIEW

File: lib/employee.rb [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+              :name,
+              :role,
+              :start_date,
+              :end_date
+
+  def initialize(employee_id, name, role, start_date, end_date)
+    @id           = employee_id.to_i
+    @name         = name
+    @role         = role
+    @start_date   = DateHandler.string_to_date(start_date)
```
Comment (iandouglas) [inline lib/employee.rb]: nice!

File: test/company_test.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    company = Company.new
+    filename = './data/bad_timesheets.csv'
+  end
+
+  def test_it_can_find_employee_by_id
+    company = Company.new
+    filename = './data/employees.csv'
+    result = company.load_employees(filename)
+
+    assert_instance_of Employee, company.find_employee_by_id(1)
```
Comment (iandouglas) [inline test/company_test.rb]: calling `company.find_employee_by_id` over and over is redundant, just do something like:
`employee = company.find_employee_by_id(1)` and use `employee.role` etc below.

File: lib/company.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+              :timesheets
+
+  def initialize
+    @employees  = []
+    @projects   = []
+    @timesheets = []
+  end
+
+  def load_employees(filename)
+    CSV.foreach(filename, headers: false) do |data|
```
Comment (mikedao) [inline lib/company.rb]: I think I'd like to have seen your CSV stuff broken out to another class, we've got a bunch of repetition in here.

File: lib/company.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      @status = returns(data, 5)
+      if @status[:success] == true && @status[:error].nil?
```
Comment (mikedao) [inline lib/company.rb]: Good opportunity to use a predicate method here.

**Pull request closed without merging**